### DISCUSSION TITLE
Option to add state store to topology

### DISCUSSION THREAD

**Discussion by ppilev** [LGouellec/streamiz#274] [General]
Hello @LGouellec 

I've seen in some snippets written in Scala that a store builder can be added to the topology so later you can use that sore in a processor.

``` Scala
val storeSupplier = Stores.inMemoryKeyValueStore("store-name")
val storeBuilder = Stores.keyValueStoreBuilder(storeSupplier, Serdes.X, Serdes.Y)

// Create Stream builder
val builder = new StreamsBuilder
builder.addStateStore(storeBuilder)

```
Is it somehow possible to do that with Streamiz?

Thanks,

**Comment by LGouellec** [comment]
Hi @ppilev , 

Sorry for the long delay. No it's not possible for now, you have to use the processor API with the processor builder directly. 

Thanks,

**Reply by ppilev** [reply]
hi @LGouellec,

The requested functionality has been added with this PR
https://github.com/LGouellec/kafka-streams-dotnet/pull/284

Could you please check it?

Regards,
